x-pack/plugins/apm/common/rules/apm_rule_types.ts

```diff
@@ -40,22 +40,36 @@ const THRESHOLD_MET_GROUP: ActionGroup<ThresholdMetActionGroupId> = {
   }),
 };
 
-const formatGroup = (group: string) => (group ? ` for ${group}` : '');
+const notDefinedLabel = i18n.translate('xpack.apm.field.notDefinedLabel', {
+  defaultMessage: 'not defined',
+});
+
+const getValueForNotDefinedField = (key: string) => {
+  return `${key
+    .replace('_NOT_DEFINED', '')
+    .replaceAll('_', ' ')
+    .toLowerCase()} ${notDefinedLabel}`;
+};
+
+const formatGroupKey = (groupKey: string[]) =>
+  ` for ${groupKey
+    .map((key) =>
+      key.includes('NOT_DEFINED') ? getValueForNotDefinedField(key) : key
```

Comment:
We already have some helpers in `x-pack/plugins/apm/common/environment_filter_values.ts`.

```suggestion
      key === ENVIRONMENT_NOT_DEFINED_VALUE ? getEnvironmentLabel(key) : key
```

Btw. we should avoid doing `key.includes('NOT_DEFINED')` because it makes it harder to reason about. If we want custom formatting we should do it for specific values (eg `ENVIRONMENT_NOT_DEFINED_VALUE`)

Reply:
This is to address all group keys that are not defined. I think it is better to generalize it than checking each key individually. If we have more group keys in future, then checking `NOT_DEFINED` is more readable, IMO.
I didn't use `environment_filter_values.ts` because the naming was specific to `environment`, whereas I have applied this to all group keys.